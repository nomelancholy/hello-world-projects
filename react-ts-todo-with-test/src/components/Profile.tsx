import React, { useState } from "react";

interface Props {
  userName: string;
  name: string;
}

const Profile = ({ userName, name }: Props) => {

    const [inputText, setInputText] = useState('')

 const handleChange = (e:any) => {
     setInputText(e.target.value)
 }


  return (
    <div>
        <input type="text" onChange={handleChange} value={inputText} placeholder="anything"/>
      <span>
        <b>{userName}</b>
      </span>
      <span>{name}</span>
    </div>
  );
};

export default Profile;
