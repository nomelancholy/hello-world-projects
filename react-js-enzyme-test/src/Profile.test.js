import React from 'react';
import { mount } from 'enzyme';
import Profile from './Profile';

describe('<Profile />', () => {
    it('matches snapshot', () => {
        const wrapper = mount(<Profile username="Take Knowledge" name="최재욱" />);
        expect(wrapper).toMatchSnapshot();
    });
    it('renders username and name', () => {
        const wrapper = mount(<Profile username="Take Knowledge" name="최재욱" />);

        expect(wrapper.props().username).toBe('Take Knowledge');
        expect(wrapper.props().name).toBe('최재욱');

        const boldElement = wrapper.find('b');
        expect(boldElement.contains('Take Knowledge')).toBe(true);
        const spanElement = wrapper.find('span');
        expect(spanElement.text()).toBe('(최재욱)')
    });
});