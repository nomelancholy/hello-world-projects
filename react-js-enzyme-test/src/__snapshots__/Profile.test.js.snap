// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> matches snapshot 1`] = `
<Profile
  name="최재욱"
  username="Take Knowledge"
>
  <div>
    <b>
      Take Knowledge
    </b>
     
    <span>
      (
      최재욱
      )
    </span>
  </div>
</Profile>
`;
